### PR TITLE
feat(viewer): gate check-in badge on checkInEnabled (mp-wkm)

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

- Extracts `buildRoster(competitions)` helper that deduplicates player lists across competitions while masking `checkedIn` to `false` for any competition where `checkInEnabled` is `false`
- Replaces two identical inline `useMemo` roster constructions (in `ViewerHome` and `WatchlistPanel`) with calls to `buildRoster`, fixing both rendering sites — `SinglePlayerPicker` and the watchlist chip — at once
- When a player appears in multiple competitions, `checkedIn=true` wins if ANY check-in-enabled competition has them checked in (correct semantics for a cross-competition roster)
- Adds `viewer_checkin.test.jsx` with 9 unit tests covering masking, preservation, multi-comp merging, deduplication, upgrade semantics, missing-field coercion, and edge cases

## Motivation

Defense-in-depth: `viewer.jsx` was rendering `✓ Checked in` badges unconditionally based on `pRecord.checkedIn`. When `checkInEnabled` is `false`, no admin path can write `checkedIn=true` today — but the view should gate on both values so the rendering contract stays explicit. Discovered by `/review` on PR #136 (mp-6nq) and filed as mp-wkm.

## Test plan

- [x] `npm test` in `web-mobile/` — 854 tests pass (28 files)
- [x] New `viewer_checkin.test.jsx` — 9 tests all green
- [x] Manual: start app, create a competition with `checkInEnabled=false`, check in a participant via the API, open the viewer — no check-in badge appears (verified with Alice Smith in setup-test)
- [x] Manual: repeat with `checkInEnabled=true` — badge appears correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: web-mobile/js/viewer.jsx [hunk 1]
```diff
@@ -262,6 +262,24 @@ function buildWatchlistUpcoming(watched, allMatches, max = WATCHED_UPCOMING_MAX)
...
+  const map = new Map();
+  (competitions || []).forEach((c) => {
+    (c.players || []).forEach((p) => {
+      if (!p || !p.id) return;
+      const checkedIn = !!c.checkInEnabled && p.checkedIn;
+      const existing = map.get(p.id);
+      if (!existing) {
+        map.set(p.id, { ...p, checkedIn });
+      } else if (checkedIn && !existing.checkedIn) {
+        map.set(p.id, { ...existing, checkedIn: true });
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/viewer.jsx:276]: Good catch —  was dropped when the function was simplified (the original implementation had it). Fixed in 08cc4540: added  back so the result is always a strict boolean. Also added a regression test: player with missing  field +  →  (was yielding  before the fix).
